labs/b9.md

```diff
@@ -1,159 +1,539 @@
+messages, Git will bring up another text editor for you to modify the commit
+message. Because you squashed, Git made a multi-line commit message by combining
+the three individual commit messages from before you squashed. Let's add a
+useful one-line summary, "Add dice roll feature", at the top of the file. You
+can also adjust the formatting to your liking; I like adding bullet points to
+the individual commit messages.
 
-    Added my new feature
+```
+Add dice roll feature
 
-commit cbe253d523c12f05595b8d84d7195cb728718967
-Author: William Porr <wporr@nvidia.com>
-Date:   Mon Oct 28 18:55:51 2019 -0700
+# This is a combination of 3 commits.
+# This is the 1st commit message:
 
-    Added correct output
+- Add -s flag for number of sides on a die
 
-commit 6e14391676dc6213cc6f4e186dcb88f3edf84994 (master)
-Author: William Porr <wporr@nvidia.com>
-Date:   Mon Oct 28 18:58:00 2019 -0700
+# This is the commit message #2:
 
-    Added changed output
+- Add dice rolling logic and output dice sum and sequence
 
-commit b9a2169e7d53d7a30ae933e09aeb8e8245117c0c
-Author: William Porr <wporr@nvidia.com>
-Date:   Mon Oct 28 18:43:21 2019 -0700
+# This is the commit message #3:
 
-    Added b9 code
+- Restrict input range for dice iterations and sides
 ```
 
-Now that our branch has a clean history, we can merge it with the master branch.
-This is where you would normally submit a pull request on github or some other
-shared git interface, but we're just going to merge the branch ourselves.
-Switch to the master branch with `git checkout master` and run `git merge
-my_branch` to merge your branch into master.
+> Git and other tools use the first line of a commit message as a shortened
+> version for certain display purposes. For example, try running `git log
+> --oneline`. As another example, when Github displays the most recent commit
+> message next to files and folders in the directory view, the first line of the
+> message is shown.
 
+Save the file and exit. The rebase is now complete! Type `git log` and view your
+new squashed commit. If you didn't already merge your three individual commits
+into `master`, you can run `git merge master` to merge in your combined commit. 
 
-### Cleaning up
-Now that your change is rebased, you can delete your branch with `git
-branch -D my_branch`
+```
+commit c0fa61ad59a76635701813472bfbc1c7f36bb857 (HEAD -> dice)
+Author: Edric Xiang <email redacted>
+Date:   Wed Aug 12 19:00:29 2020 -0700
 
-## Questions
+    Add dice roll feature
 
-1. What caused the merge conflict in the git exercises you did?
+    - Add -s flag for number of sides on a die
 
-2. Why does git require us to manually intervene in merge conflicts?
+    - Add dice rolling logic and output dice sum and sequence
 
-3. What command would you use to sync a folder ~/Downloads/Linux_ISOs to the folder /usr/local/share/Calendar, while preserving file metadata? (hint: use rsync)
+    - Restrict input range for dice iterations and sides
```

Comment:
Add a submittable question at the end of part 1 so that we can verify that they actually followed the instructions in this section.